app/models/discount-code.js

```diff
@@ -25,12 +26,14 @@ export default ModelBase.extend({
   discountUrl   : attr('string'),
   validTill     : attr('moment', { defaultValue: () => moment.tz(detectedTimezone).add(1, 'months').startOf('day') }),
   usedFor       : attr('string'),
-  isActive      : attr('boolean', { defaultValue: true }),
   createdAt     : attr('moment'),
 
-  tickets : hasMany('ticket'),
-  orders  : hasMany('order'),
-  event   : belongsTo('event', {
+  tickets  : hasMany('ticket'),
+  orders   : hasMany('order'),
+  isActive : computed('validTill', function() {
+    return (new Date() <= new Date(this.get('validTill')));
```

Comment:
no -- you can't overwrite a property coming from the server - it's not the convention. Define a new propery called status. Calculate using both: isActive && <validTill condition> and then according to it decide the status of discount code. Make three statuses: active, inactive and expired. Active/inactive were created for the intention that an organizer can go at any time make a discount code inactive to prevents its usage without changing anything else. Valid till had nothing to do with it.

Reply:
Ok. So what I plan to do is check if the <validTill condition> is satisfied or not. If it is satisfied, then `isActive` comes in play and the status is either `Active` or `Inactive` depending on `isActive`.
If <validTill condition> is not satisfied, then the status is marked as `Expired`. Is this fine?

Reply:
yup @shreyanshdwivedi sounds good.